simulators/nav/src/components/perception/obstacle_detector.ts

```diff
@@ -279,12 +288,20 @@ export default class ObstacleDetector {
         return null;
       }
     }
-
-    return {
-      detected: false, /* deprecated so always false */
-      distance: this.obsDist, /* Will be -1 if okay to go straight ahead (i.e. bearing = 0) */
-      bearing: calcRelativeBearing(this.zedOdom.bearing_deg, angle)
-    };
+    if (this.obsDist >= 0) {
+      return {
+        detected: true, /* deprecated so always false */
+        distance: this.obsDist, /* Will be -1 if okay to go straight ahead (i.e. bearing = 0) */
+        bearing: calcRelativeBearing(this.zedOdom.bearing_deg, angle)
+      };
+    }
+    else {
+      return {
+        detected: false, /* deprecated so always false */
+        distance: this.obsDist, /* Will be -1 if okay to go straight ahead (i.e. bearing = 0) */
+        bearing: calcRelativeBearing(this.zedOdom.bearing_deg, angle)
+      };
```

Comment:
I don't think you need to do this twice. You are only affected by the version of these that is will eventually be published via the LCM, right? Could be this one or the other, but not both is my understanding.